packages/bento-design-system/src/TextField/createTextField.tsx

```diff
@@ -31,34 +38,71 @@ export function createTextField(config: InputConfig, { Field }: { Field: FieldTy
       inputRef
     );
 
+    const [showPassword, setShowPassword] = useState(false);
+    const passwordIcon = useMemo(
+      () =>
+        showPassword
+          ? config.passwordHideIcon ?? IconEye
+          : config.passwordShowIcon ?? IconEyeClosed,
+      [showPassword]
+    );
+    const iconProps: IconProps = useMemo(() => ({ size: config.passwordIconSize || 24 }), []);
+
+    const type = useMemo(() => {
+      if (props.type === "password") {
+        return showPassword ? "text" : "password";
+      } else {
+        return props.type || "text";
+      }
+    }, [props.type, showPassword]);
+
```

Comment:
I would remove all the `useMemo` uses here: they make the code a bit harder to read, and I don't think there is any tangible benefit (the computations a very cheap)